src/libpriv/rpmostree-core.c

```diff
@@ -2093,6 +2095,7 @@ rpmostree_context_import_jigdo (RpmOstreeContext *self,
 
     self->async_dnfstate = hifstate;
     self->async_running = TRUE;
+    self->async_cancellable = g_cancellable_new ();
```

Comment:
Sorry, I'm still not clear on how this works exactly. We create a new `GCancellable` here, but when do we actually consult it? We're still passing our own `cancellable` to `rpmostree_importer_run_async` so the `g_cancellable_set_error_if_cancelled` importer fixup you added is not actually using this, right? So, aren't we still going to end up trying to import all the pkgs even if the very first one failed? Or should we be passing `self->async_cancellable` to `rpmostree_importer_run_async` down below?

Reply:
No you're totally right, it doesn't work 😉.  I was throwing code at the wall here a bit.  Let's do the obvious thing here and reuse the source cancellable, which is what ostree-pull does too.

Reply:
Pushed a fixup :arrow_down: which I tested by adding:

```
if (g_random_boolean ())
  return glnx_throw (...)
```
in the `relabel_in_thread()` code.